recipe-app/View/RecipeListViewController.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ViewController.swift
+//  recipe-app
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import UIKit
+
+final class RecipeListViewController: UIViewController {
+    
+    var recipeListViewModel = RecipeListViewModel()
+    
+    let searchBar: UISearchBar = {
+        let searchBar = UISearchBar()
+        searchBar.placeholder = "Search"
+        searchBar.tintColor = .blue
+        searchBar.searchBarStyle = .default
+        return searchBar
+    }()
+    
+    let collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.register(CustomCollectionViewCell.self, forCellWithReuseIdentifier: CustomCollectionViewCell.identifier)
+        return collectionView
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.view.backgroundColor = .systemBlue
+        
+        recipeListViewModel.searchRecipe("") { result in
+            switch result {
+            case .success(_):
+                self.collectionView.reloadData()
+                self.recipeListViewModel.fetchImages(""){
+                    self.collectionView.reloadData()
+                }
+                
+            case .failure(let error):
+                print(error)
+            }
+        }
+        
+        self.setupUI()
+        self.collectionView.dataSource = self
+        self.collectionView.delegate = self
+    }
+    
+    func setupUI() {
+        
+        self.view.backgroundColor = .systemMint
+        self.view.addSubview(collectionView)
+        self.view.addSubview(searchBar)
+        
+        searchBar.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
You can set these in their corresponding components.
```suggestion
       private let collectionView: UICollectionView = {
        let layout = UICollectionViewFlowLayout()
        layout.scrollDirection = .vertical

        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
        collectionView.register(CustomCollectionViewCell.self, forCellWithReuseIdentifier: "CustomCollectionViewCell")
        collectionView.translatesAutoresizingMaskIntoConstraints =  false
        return collectionView
    }()
```